Improvement: mqtt library replacement

No-Signal opened the pull request:
Replacement of the unsupported MQTT library from Issue #761 using the ESP-MQTT library

This PR requires further testing & investigation into flash size before it can be considered ready

The initial MQTT connection always fails as the Wifi connection is not yet ready, need to implement a way to wait for Wifi to connect before starting the MQTT connection

No-Signal commented:
This version has fixes for #785 #761 #743 and adds some of the features listed in #239

dalathegreat commented on README.md:
```diff
@@ -52,17 +52,19 @@ For more examples showing wiring, see each battery types own Wiki page. For inst
+
 This video explains all the above mentioned steps:
 <https://youtu.be/_mH2AjnAjDk>
 
```
Nice work! Could your remove the mention of lib dependency, essentially remove this row from README.md

- [knolleary/pubsubclient](https://github.com/knolleary/pubsubclient) MIT-License

dalathegreat reviewed (approved):
I am approving this one to make it easier to merge later on, good luck with the LWT and QoS 1! :raised_hands:

dalathegreat commented:
@No-Signal , can you update the comment on username/password in USER_SECRETS as described in issue #738 ?

No-Signal commented:
> @No-Signal , can you update the comment on username/password in USER_SECRETS as described in issue #738 ?
> 
> EDIT: No need, I fixed this in #802

I have updated the template again as the new library allows setting a blank string for no authentication which should be easier for people to update
